src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/ServiceProviderEngineScope.cs

```diff
@@ -66,16 +67,17 @@ internal object CaptureDisposable(object service)
                         // sync over async, for the rare case that an object only implements IAsyncDisposable and may end up starving the thread pool.
                         Task.Run(() => ((IAsyncDisposable)service).DisposeAsync().AsTask()).GetAwaiter().GetResult();
                     }
+
                     ThrowHelper.ThrowObjectDisposedException();
-                }
 
-                if (_disposables == null)
-                {
-                    _disposables = new List<object>();
+                    return service;
```

Comment:
?? Huh?

Reply:
Yea, this is kinda not required. It got added due to some refactoring and un-refactoring

Reply:
Can we remove it?

Reply:
Yep.